src/sagemaker/cli/compatibility/v2/modifiers/serde.py

```diff
@@ -0,0 +1,413 @@
+OLD_CLASS_NAMES = set(OLD_CLASS_NAME_TO_NEW_CLASS_NAME.keys())
+
+OLD_OBJECT_NAMES = set(OLD_OBJECT_NAME_TO_NEW_CLASS_NAME.keys())
+
+
+class SerdeConstructorRenamer(Modifier):
+    """A class to rename SerDe classes."""
+
+    def node_should_be_modified(self, node):
+        """Checks if the ``ast.Call`` node instantiates a SerDe class.
+
+        This looks for the following calls:
+
+        - ``sagemaker.predictor._CsvSerializer``
+        - ``sagemaker.predictor._JsonSerializer``
+        - ``sagemaker.predictor._NpySerializer``
+        - ``sagemaker.predictor._CsvDeserializer``
+        - ``sagemaker.predictor.BytesDeserializer``
+        - ``sagemaker.predictor.StringDeserializer``
+        - ``sagemaker.predictor.StreamDeserializer``
+        - ``sagemaker.predictor._NumpyDeserializer``
+        - ``sagemaker.predictor._JsonDeserializer``
+        - ``sagemaker.amazon.common.numpy_to_record_serializer``
+        - ``sagemaker.amazon.common.record_deserializer``
+
+        Args:
+            node (ast.Call): a node that represents a function call. For more,
+                see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            bool: If the ``ast.Call`` instantiates a SerDe class.
+        """
+        return matching.matches_any(node, OLD_CLASS_NAME_TO_NAMESPACES)
+
+    def modify_node(self, node):
+        """Modifies the ``ast.Call`` node to use the classes for SerDe
+        available in version 2.0 and later of the Python SDK:
+
+        - ``sagemaker.serializers.CSVSerializer``
+        - ``sagemaker.serializers.JSONSerializer``
+        - ``sagemaker.serializers.NumpySerializer``
+        - ``sagemaker.deserializers.CSVDeserializer``
+        - ``sagemaker.deserializers.BytesDeserializer``
+        - ``sagemaker.deserializers.StringDeserializer``
+        - ``sagemaker.deserializers.StreamDeserializer``
+        - ``sagemaker.deserializers.NumpyDeserializer``
+        - ``sagemaker.deserializers._JsonDeserializer``
+        - ``sagemaker.amazon.common.RecordSerializer``
+        - ``sagemaker.amazon.common.RecordDeserializer``
+
+        Args:
+            node (ast.Call): a node that represents a SerDe constructor.
+
+        Returns:
+            ast.Call: a node that represents the instantiation of a SerDe object.
+        """
+        class_name = node.func.id if isinstance(node.func, ast.Name) else node.func.attr
+        new_class_name = OLD_CLASS_NAME_TO_NEW_CLASS_NAME[class_name]
+
+        # We don't change the namespace for Amazon SerDe.
+        if class_name in OLD_AMAZON_CLASS_NAMES:
+            if isinstance(node.func, ast.Name):
+                node.func.id = new_class_name
+            elif isinstance(node.func, ast.Attribute):
+                node.func.attr = new_class_name
+            return node
+
+        namespace_name = NEW_CLASS_NAME_TO_NAMESPACES[new_class_name][0]
+        subpackage_name = namespace_name[namespace_name.find(".") + 1 :]
```

Comment:
(optional) you could also do:

```suggestion
        subpackage_name = namespace_name.split(".")[1]
```